### PR TITLE
Replace class 0 SMS model with separate D-Bus interface (+ small cleanups)

### PR DESCRIPTION
**Pull request by pvuorela**
See commits.

### CODE REVIEW

File: src/textchannellistener.cpp [hunk 1]
```diff
@@ -690,13 +682,11 @@ void TextChannelListener::handleMessages()
...
-                // just ack message, expunge would be called
-                // as soon as user reads message
-                DEBUG() << __FUNCTION__ << "Adding class 0 sms";
+                DEBUG() << __FUNCTION__ << "Handling class 0 sms";
                 processedMessages << message;
-                classZeroSMSModel()->addEvent(event,true);
-                m_EventTokens.insertMulti(event.id(), event.messageToken());
                 nManager->playClass0SMSAlert();
+                nManager->requestClass0Notification(event);
+                expungeMessage(event.messageToken());
```
Comment (pvuorela) [inline src/textchannellistener.cpp]: expungeMessage() looks suspicious, btw. Sets expunge timer only if something was already added to m_expungeTokens. Am I missing something?

**Comment by thp** [comment]
Needs rebase? Let me know if you want me do to a patch review on this one.

**Comment by pvuorela** [comment]
Rebased to avoid version bump conflict.

All review always appreciated, but would like at least @special to approve.

**Comment by special** [comment]
Simple solution, LGTM.

We should get rid of classzerosmsmodel in libcommhistory once it's not used anymore.

**Comment by thp** [comment]
LGTM too, thanks for getting rid of the Qt 4 cruft along the way :+1:

**Comment by pvuorela** [comment]
libcommhistory side:
https://github.com/nemomobile/libcommhistory/pull/104

**Comment by jpetrell** [comment]
LGTM

**Comment by pvuorela** [comment]
Rebased to fix version bump conflict.

**Pull request merged**